### PR TITLE
#571 invoice service charges

### PR DESCRIPTION
**Pull request by joshxg**
Starts #571 

- This is more of a cleanup to introduce service items rather than implementing service charges completely.
- Added `useServiceItems` and `useStockItems`. I've left `useItems` for now as the list view is using it. Though, it seems that mSupply is filtering these out unless you ... "turn off exclusion of hidden items" - what a button! - but thought to leave that until concentrating more on the item page/pages

### CODE REVIEW

File: packages/inventory/src/Stocktake/DetailView/modal/StocktakeLineEdit/StocktakeLineEditForm.tsx [hunk 1]
```diff
@@ -23,23 +26,23 @@ export const StocktakeLineEditForm: FC<StocktakeLineEditProps> = ({
...
-              const itemAlreadyInStocktake = items?.some(
-                ({ item }) => itemToCheck.id === item?.id
-              );
-              return !itemAlreadyInStocktake;
-            }}
+            extraFilter={
+              disabled
+                ? undefined
+                : item => !items?.some(({ id }) => id === item.id)
+            }
```
Comment (joshxg) [inline packages/inventory/src/Stocktake/DetailView/modal/StocktakeLineEdit/StocktakeLineEditForm.tsx:45]: Changed to `StockItemSearchInput` and offroad fixed the warning about the value being invalid (when editing a line, it was filtering out the item but setting it as the value).

I did this for all request, stocktake, outbound, inbound but not response as that item is non-editable

File: packages/invoices/src/OutboundShipment/DetailView/OutboundServiceLineEdit/OutboundServiceLineEdit.tsx [hunk 2]
```diff
@@ -14,7 +14,7 @@ export const OutboundServiceLineEdit = ({
 
   return (
     <Modal title="Service Charges">
-      <span>:)</span>
+      <ServiceItemSearchInput width={300} onChange={() => {}} />
```
Comment (joshxg) [inline packages/invoices/src/OutboundShipment/DetailView/OutboundServiceLineEdit/OutboundServiceLineEdit.tsx:17]: Just added this for now to have a looksie - need to setup a nice msw system with storybook 🤔

File: packages/system/src/Item/Components/ServiceItemSearchInput/ServiceItemSearchInput.tsx [hunk 3]
```diff
@@ -0,0 +1,68 @@
+import React, { FC } from 'react';
```
Comment (joshxg) [inline packages/system/src/Item/Components/ServiceItemSearchInput/ServiceItemSearchInput.tsx:1]: A little bit different to the stock input. Specifically no available stock. I also removed the code... maybe that's a mistake but it looked nicer 😁

File: packages/system/src/Item/ListView/ListView.tsx [hunk 4]
```diff
@@ -6,11 +6,11 @@ import {
   useColumns,
   createTableStore,
 } from '@openmsupply-client/common';
-import { useItemListView, ItemRowFragment } from '../api';
+import { useItems, ItemRowFragment } from '../api';
```
Comment (joshxg) [inline packages/system/src/Item/ListView/ListView.tsx:9]: renamed `useItemListView` -> `useItems` 🤔

File: packages/system/src/Item/api/api.ts [hunk 5]
```diff
@@ -1,10 +1,10 @@
 import {
+  ItemNodeType,
```
Comment (joshxg) [inline packages/system/src/Item/api/api.ts:2]: I'm somewhaaat tempted to make the folder structure closer to `item/service`, `item/stock`, `item` and putting the specific things in each folder but it seemed OTT

File: packages/system/src/Item/api/hooks/useServiceItems/useServiceItems.ts [hunk 6]
```diff
@@ -0,0 +1,11 @@
+import { useQuery, useQueryParams } from '@openmsupply-client/common';
+import { useItemApi } from './../useItemApi';
+
+export const useServiceItems = () => {
+  const api = useItemApi();
+  const queryParams = useQueryParams({ initialSortBy: { key: 'name' } });
+
+  return useQuery(api.keys.paramList(queryParams), () =>
+    api.get.serviceItems(queryParams)
+  );
```
Comment (joshxg) [inline packages/system/src/Item/api/hooks/useServiceItems/useServiceItems.ts:10]: Maybe I should have just used the same api calls with filters. However I think the queries for stock items and service items will change in the future for different reasons and will cause headaches

Comment (mark-prins) [reply packages/system/src/Item/api/hooks/useServiceItems/useServiceItems.ts:10]: The separation is helpful - some with pagination, some not, different fragments etc

**Review by mark-prins — APPROVED** [review]
🎉  service items!

File: packages/system/src/Item/api/operations.graphql [hunk 7]
```diff
@@ -121,8 +128,15 @@ query itemsListView(
...
-  items(storeId: $storeId) {
+query itemsWithStats(
+  $first: Int
+  $offset: Int
+  $storeId: String!
+  $key: ItemSortFieldInput
+  $isDesc: Boolean
+  $filter: ItemFilterInput
+) {
+  items(storeId: $storeId, sort: { key: $key, desc: $isDesc }) {
```
[the remark this answers is not in the document]
Comment (joshxg) [reply packages/system/src/Item/api/operations.graphql]: oh crap

**Pull request closed without merging**